src/openms_gui/source/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.cpp

```diff
@@ -126,34 +144,39 @@ namespace OpenMS
   {
     Param p;
 
-    bool peptide_input = ui_->seq_type->currentText() == "Peptide";
-
-    // add check boxes to parameters, i.e. ion types
-    for (size_t i = 0; i < int(CheckBox::NUMBER_OF_CHECK_BOXES); ++i)
+    if (seq_type_ != "Metabolite") // no ions for metabolite input
```

Comment:
using a string seems dangerous.
If someone changes the content of the dropdown, everything breaks.
Why not make this an enum?!